src/quote/flashmint/hyeth/provider.ts

```diff
@@ -0,0 +1,103 @@
+import { BigNumber } from '@ethersproject/bignumber'
+
+import { WETH } from 'constants/tokens'
+import { QuoteProvider, QuoteToken } from 'quote/interfaces'
+import { SwapQuoteProvider } from 'quote/swap'
+import { slippageAdjustedTokenAmount, SwapData } from 'utils'
+
+import { ComponentQuotesProvider } from './component-quotes'
+import { getRequiredComponents } from './issuance'
+import {
+  getComponentsSwapData,
+  getEthToInputOutputTokenSwapData,
+  getInputTokenToEthSwapData,
+} from './swap-data'
+
+export interface FlashMintHyEthQuoteRequest {
+  isMinting: boolean
+  inputToken: QuoteToken
+  outputToken: QuoteToken
+  indexTokenAmount: bigint
+  slippage: number
+}
+
+export interface FlashMintHyEthQuote {
+  indexTokenAmount: bigint
+  inputOutputTokenAmount: bigint
+  // Represents `swapDataEthToComponent` for minting
+  // and `swapDataComponentToEth` for redeeming
+  componentsSwapData: SwapData[]
+  // Used only for minting w/ ERC-20 tokens
+  swapDataInputTokenToEth: SwapData | null
+  // Represents `swapDataEthToInputToken` for minting w/ ERC-20 token
+  // and `swapDataEthToOutputToken` for redeeming to ERC-20 token
+  swapDataEthToInputOutputToken: SwapData | null
+}
+
+export class FlashMintHyEthQuoteProvider
+  implements QuoteProvider<FlashMintHyEthQuoteRequest, FlashMintHyEthQuote>
+{
+  constructor(
+    private readonly rpcUrl: string,
+    private readonly swapQuoteProvider: SwapQuoteProvider
+  ) {}
+
+  async getQuote(
+    request: FlashMintHyEthQuoteRequest
+  ): Promise<FlashMintHyEthQuote | null> {
+    const { indexTokenAmount, inputToken, isMinting, outputToken, slippage } =
+      request
+    const componentsSwapData = getComponentsSwapData(isMinting)
+    // Only relevant for minting ERC-20's
+    const swapDataInputTokenToEth = isMinting
+      ? getInputTokenToEthSwapData(inputToken)
+      : null
+    const inputOutputToken = isMinting ? inputToken : outputToken
+    const swapDataEthToInputOutputToken =
+      getEthToInputOutputTokenSwapData(inputOutputToken)
+
+    const { components, positions } = await getRequiredComponents(
+      request,
+      this.rpcUrl
+    )
+
+    if (componentsSwapData.length !== components.length) return null
+
+    // Mainnet only for now
+    const chainId = 1
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const wethAddress = WETH.address!
+    const quoteProvider = new ComponentQuotesProvider(
+      chainId,
+      slippage,
+      wethAddress,
+      this.rpcUrl,
+      this.swapQuoteProvider
+    )
+    const quoteResult = await quoteProvider.getComponentQuotes(
+      components,
+      positions,
+      isMinting,
+      inputToken,
+      outputToken
+    )
+    if (!quoteResult) return null
+    const inputOutputTokenAmount = slippageAdjustedTokenAmount(
+      BigNumber.from(quoteResult.inputOutputTokenAmount.toString()),
+      isMinting ? inputToken.decimals : outputToken.decimals,
+      slippage,
+      isMinting
+    )
```

Comment:
@pblivin0x we used this slippage adjusting function for ZeroEx as well. it seems to makes sense here as otherwise mint quotes can revert.

Reply:
it makes sense to use some slippage adjustments, but they may be better placed in the integrations

- Instadapp - implementation need to replace preview method https://github.com/IndexCoop/flash-mint-sdk/pull/64#discussion_r1631752568
- Across - implementation already applies a small rounding error on deposits https://github.com/IndexCoop/flash-mint-sdk/pull/64#discussion_r1631749292
- Pendle - its known that large deposits/withdrawals will move the exchangeRate() oracle, and this created issues in FlashMintHyETH and auction bots. here are some notes on handling https://github.com/IndexCoop/flash-mint-sdk/pull/64#discussion_r1631754303 it will probably be best check previews before applying a similar inflation factor 

with these itemized adjustments made, it may not be necessary to do a overall slippage adjustment here

Reply:
Individual changes have been made but we still might wanna watch slippage for a bit.